### PR TITLE
Fixes and updates to help.md

### PR DESCRIPTION
**Pull request by ohthetrees**
The demonstration of lists inside of block quotes wasn't rendering as intended, so I fixed that. Also made a few other small improvements while I was at it.

### CODE REVIEW

**Comment by uranusjr** [comment]
LGTM, but could you rephrase the commit message to follow the rules specified [here](https://github.com/uranusjr/macdown/blob/master/CONTRIBUTING.md#commit-messages)? Thanks. :)

**Comment by ohthetrees** [comment]
I've confused myself. I'll close this and make a new pull request.
